kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,278 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.freedesktop.dbus.errors.NotSupported;
+import org.junit.Test;
+
+public class NetworkPropertiesTest {
+
+	NetworkProperties netProps;
+	Map<String, Object> properties;
+	Optional<?> optResult;
+	String stringResult;
+	List<String> stringListResult;
+	Map<String, Object> resultMap;
+
+	Boolean hasNullPointExceptionBeenThrown = false;
+
+	@Test
+	public void shouldFailWhenNull() {
+		givenNetworkPropertiesBuiltWith(null);
+		givenNetworkPropsIsCreated();
+		thenAnNullPointerExceptionOccured();
+	}
+
+	@Test
+	public void shouldReturnPropertiesWithGetProperties() {
+		givenTheMapWith("testKey1", "testString1");
+		givenNetworkPropsIsCreated();
+		whenGetPropertiesIsCalled();
+		thenResultEquals(Collections.singletonMap("testKey1", "testString1"));
+	}
+
+	@Test
+	public void shouldReturnStringWithGet() {
+		givenTheMapWith("testKey1", "testString1");
+		givenNetworkPropsIsCreated();
+		whenGetIsCalledWith("testKey1", String.class);
+		thenResultEquals("testString1");
+	}
+	
+	@Test
+	public void shouldReturnStringWithGetNull() {
+		givenTheMapWith("testKey1", null);
+		givenNetworkPropsIsCreated();
+		whenGetIsCalledWith("testKey1", String.class);
+		thenResultEquals((String) null);
+	}
+	
+	@Test
+	public void shouldReturnStringWithGetMissing() {
+		givenTheMapWith("testKey1", null);
+		givenNetworkPropsIsCreated();
+		whenGetIsCalledWith("testKey1-nonExistant", String.class);
+		thenResultEquals((String) null);
+	}
+
+	@Test
+	public void shouldReturnStringWithGetOpt() {
+		givenTheMapWith("testKey1", "testString1");
+		givenNetworkPropsIsCreated();
+		whenGetOptIsCalledWith("testKey1", String.class);
+		thenResultEquals(Optional.of("testString1"));
+	}
+
+	@Test
+	public void shouldReturnEmptyGetOptEmpty() {
+		givenTheMapWith("testKeyEmpty", "");
+		givenNetworkPropsIsCreated();
+		whenGetOptIsCalledWith("testKeyEmpty", String.class);
+		thenResultEquals(Optional.empty());
+	}
+
+	@Test
+	public void shouldReturnEmptyGetOptNull() {
+		givenTheMapWith("testKeyNull", null);
+		givenNetworkPropsIsCreated();
+		whenGetOptIsCalledWith("testKeyNull", String.class);
+		thenResultEquals(Optional.empty());
+	}
+
+	@Test
+	public void shouldReturnEmptyGetOptEmptyKey() {
+		givenTheMapWith("testKeyNull", null);
+		givenNetworkPropsIsCreated();
+		whenGetOptIsCalledWith("", String.class);
+		thenResultEquals(Optional.empty());
+	}
+
+	@Test
+	public void shouldReturnStringListGetStringListWithCommaSeperatedString() {
+		givenTheMapWith("testKey-comma-seperated", "commaSeperated1,commaSeperated2,commaSeperated3");
+		givenNetworkPropsIsCreated();
+		whenGetStringListIsCalledWith("testKey-comma-seperated");
+		thenResultEquals(List.of("commaSeperated1", "commaSeperated2", "commaSeperated3"));
```

Comment:
This is weird... `List.of` is a method introduced in Java9... it doesn't compile for me. Are your IDE settings correct?

Replace every `List.of` with `Arrays.asList`

```suggestion
		thenResultEquals(Arrays.asList("commaSeperated1", "commaSeperated2", "commaSeperated3"));
```